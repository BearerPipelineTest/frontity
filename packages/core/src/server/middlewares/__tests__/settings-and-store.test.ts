import { settingsAndStore } from "../settings-and-store";
import { fakeNext } from "./__utilities__/fake-next";

jest.mock("@frontity/file-settings", () => ({
  getSettings: jest.fn().mockReturnValue(async () => {
    await Promise.resolve();
  }),
}));

jest.mock("../../store", () =>
  jest.fn().mockReturnValue({
    actions: [],
    libraries: [],
  })
);

describe("settingsAndStore", () => {
  it("should define settins and store", async () => {
    const packages: any = {};
    const ctx: any = { state: {}, href: "/", query: { frontity_name: "test" } };

    // Create the middleware
    const middleware = settingsAndStore(packages);

    // Call the middleware
    await middleware(ctx, fakeNext);

    expect(ctx.state.store).toMatchSnapshot();
    expect(ctx.state.settings).toMatchSnapshot();
  });
});
