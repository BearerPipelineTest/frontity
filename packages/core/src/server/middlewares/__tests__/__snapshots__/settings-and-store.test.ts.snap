// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settingsAndStore should define settins and store 1`] = `
Object {
  "actions": Array [],
  "libraries": Object {
    "frontity": Object {
      "App": [Function],
      "render": [Function],
      "template": [Function],
    },
  },
}
`;

exports[`settingsAndStore should define settins and store 2`] = `[Function]`;
